lib/payment/payment_service.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:coffeecard/data/repositories/v2/purchase_repository.dart';
+import 'package:coffeecard/generated/api/coffeecard_api_v2.swagger.swagger.dart';
+import 'package:coffeecard/service_locator.dart';
+import 'package:coffeecard/widgets/popup_card.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+enum InternalPaymentType {
+  mobilePay,
+  applePay,
+}
+
+abstract class PaymentService {
+  factory PaymentService(InternalPaymentType type, BuildContext context) {
+    switch (type) {
+      case InternalPaymentType.mobilePay:
+        return MobilePayService(context);
+      case InternalPaymentType.applePay:
+        throw UnimplementedError();
```

Comment:
Since this is an abstract class, we can have fields. The context would be meaning full to have as a field on the abstract class not the sub types.

Reply:
this may not be as straight-forward as we think because we use the factory constructor. We may need to find a different approach